rosidl_generator_java/resource/msg.cpp.em

```diff
@@ -92,6 +98,10 @@ elif message_c_include_prefix.endswith('__result'):
     message_c_include_prefix = message_c_include_prefix[:-8]
 elif message_c_include_prefix.endswith('__feedback'):
     message_c_include_prefix = message_c_include_prefix[:-10]
+elif message_c_include_prefix.endswith('__send_goal'):
+    message_c_include_prefix = message_c_include_prefix[:-11]
+elif message_c_include_prefix.endswith('__get_result'):
+    message_c_include_prefix = message_c_include_prefix[:-12]
```

Comment:
same question here

Reply:
In this case, we do need this logic otherwise we get a compilation error. The reason is that the include prefix for a request or response message also contains the action service name. For example, the test_msgs/action/Fibonacci send goal service request message resolves to `test_msgs/action/fibonacci__send_goal__request`. The `__request` suffix is first removed above, and then these extra rules remove the `__send_goal` suffix.

Reply:
ah, I didn't notice that there were two chains of if/else (I didn't notice the `if` [here](https://github.com/ros2-java/ros2_java/blob/86b4240b17c76e162ad5f8c452b19bd5a9e36a01/rosidl_generator_java/resource/msg.cpp.em#L95)).
Makes sense.